pkg/root/trusted_root_create.go

```diff
@@ -0,0 +1,221 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package root
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"sort"
+	"time"
+
+	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+	prototrustroot "github.com/sigstore/protobuf-specs/gen/pb-go/trustroot/v1"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func (tr *TrustedRoot) constructProtoTrustRoot() error {
+	tr.trustedRoot = &prototrustroot.TrustedRoot{}
+	tr.trustedRoot.MediaType = TrustedRootMediaType01
+
+	for logID, transparencyLog := range tr.rekorLogs {
+		tlProto, err := transparencyLogToProtobufTL(*transparencyLog)
+		if err != nil {
+			return fmt.Errorf("failed converting rekor log %s to protobuf: %w", logID, err)
+		}
+		tr.trustedRoot.Tlogs = append(tr.trustedRoot.Tlogs, tlProto)
+		// ensure stable sorting of the slice
+		sort.Slice(tr.trustedRoot.Tlogs, func(i, j int) bool {
+			iTime := time.Unix(0, 0)
+			jTime := time.Unix(0, 0)
+
+			if tr.trustedRoot.Tlogs[i].PublicKey.ValidFor.Start != nil {
+				iTime = tr.trustedRoot.Tlogs[i].PublicKey.ValidFor.Start.AsTime()
+			}
+			if tr.trustedRoot.Tlogs[j].PublicKey.ValidFor.Start != nil {
+				iTime = tr.trustedRoot.Tlogs[j].PublicKey.ValidFor.Start.AsTime()
+			}
+			return iTime.Before(jTime)
+		})
+	}
+
+	for logID, ctLog := range tr.ctLogs {
+		ctProto, err := transparencyLogToProtobufTL(*ctLog)
+		if err != nil {
+			return fmt.Errorf("failed converting ctlog %s to protobuf: %w", logID, err)
+		}
+		tr.trustedRoot.Ctlogs = append(tr.trustedRoot.Ctlogs, ctProto)
+		// ensure stable sorting of the slice
+		sort.Slice(tr.trustedRoot.Ctlogs, func(i, j int) bool {
+			iTime := time.Unix(0, 0)
+			jTime := time.Unix(0, 0)
+
+			if tr.trustedRoot.Ctlogs[i].PublicKey.ValidFor.Start != nil {
+				iTime = tr.trustedRoot.Ctlogs[i].PublicKey.ValidFor.Start.AsTime()
+			}
+			if tr.trustedRoot.Ctlogs[j].PublicKey.ValidFor.Start != nil {
+				iTime = tr.trustedRoot.Ctlogs[j].PublicKey.ValidFor.Start.AsTime()
+			}
+			return iTime.Before(jTime)
+		})
+	}
+
+	for _, ca := range tr.fulcioCertAuthorities {
+		caProto, err := certificateAuthorityToProtobufCA(ca)
+		if err != nil {
+			return fmt.Errorf("failed converting fulcio cert chain to protobuf: %w", err)
+		}
+		tr.trustedRoot.CertificateAuthorities = append(tr.trustedRoot.CertificateAuthorities, caProto)
+	}
+
+	for _, ca := range tr.timestampingAuthorities {
+		caProto, err := certificateAuthorityToProtobufCA(ca)
+		if err != nil {
+			return fmt.Errorf("failed converting TSA cert chain to protobuf: %w", err)
+		}
+		tr.trustedRoot.TimestampAuthorities = append(tr.trustedRoot.TimestampAuthorities, caProto)
+	}
+
+	return nil
+}
+
+func certificateAuthorityToProtobufCA(ca CertificateAuthority) (*prototrustroot.CertificateAuthority, error) {
```

Comment:
```suggestion
func certificateAuthorityToProtobufCA(ca *CertificateAuthority) (*prototrustroot.CertificateAuthority, error) {
```

Same here on receiving via value or pointer.